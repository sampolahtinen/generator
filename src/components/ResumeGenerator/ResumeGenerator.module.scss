@import '../../styles/colors';
@import '../../styles/variables';
@import '../../styles/breakpoints';

.resumeGenerator {
  flex-wrap: wrap;
  width: 100%;
  max-width: 1095px;
  background-color: white;
  padding: 5rem 10rem;
  background-color: $white;
  box-shadow: 0 8px 14px 0 rgba(0, 0, 0, 0.2);
  z-index: $z-index_top;

  h3 {
    margin: 2rem 0;
  }

  form {
    width: 100%;
    display: flex;
    justify-content: space-between;
  }

  input {
    display: inline-block;
    background: transparent;
    width: 60%;
    border: unset;
    border-bottom: $border_bottom;
    font-size: 16px;
    height: 4.5rem;
    margin-right: 2.4rem;

    &::placeholder {
      color: $grey_200;
    }
  }

  button {
    background: $red;
    display: inline-block;
    border-radius: 2rem;
    font-size: 16px;
    width: 27rem;
    height: 4.5rem;
    color: white;
    cursor: pointer;
    border-style: none;
    transition: all 300ms ease-out;
    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.4);
    -webkit-box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.4);
    -moz-box-shadow: 0 2px 40px 0 rgba(0, 0, 0, 0.4);

    &:hover {
      background: darken($red, 5%);
    }

    &:active {
      background: darken($red, 20%);
    }
  }
}

@media screen and (max-width: $break-phone) {
  .resumeGenerator {
    padding: 2rem;
    text-align: center;
  }

  form {
    flex-direction: column;
    align-items: center;
  }

  input {
    margin: 3.2rem 0;
    width: 100%;
    text-align: center;
  }
}
