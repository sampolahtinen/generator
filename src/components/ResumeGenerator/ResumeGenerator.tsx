import React, { useState } from 'react';
import styles from './ResumeGenerator.module.scss';

interface ResumeGeneratorProps {
  onSubmit: (param?: any) => void;
}

const ResumeGenerator = ({ onSubmit }: ResumeGeneratorProps) => {
  const [githubUsername, setGithubUsername] = useState('');
  const [errorMessage, setErrorMessage] = useState('');

  const handleSubmit = (event: React.FormEvent) => {
    event.preventDefault();
    if (githubUsername) {
      setErrorMessage('');
      onSubmit(githubUsername);
    } else {
      setErrorMessage('Please provide a username.');
    }
  };

  return (
    <div className={styles.resumeGenerator}>
      <h3>Github username</h3>
      <form aria-label="Input github username">
        <label className="hidden" htmlFor="username">
          Github username input
        </label>
        <input
          id="username"
          name="username"
          placeholder="John Doe"
          type="text"
          onChange={(e: React.FormEvent<HTMLInputElement>) =>
            setGithubUsername(e.currentTarget.value)
          }
          value={githubUsername}
          autoComplete="off"
          autoFocus
        />
        <button type="submit" onClick={handleSubmit}>
          Generate
        </button>
      </form>
      <span className="highlighted-text">{errorMessage}</span>
    </div>
  );
};

export default ResumeGenerator;
