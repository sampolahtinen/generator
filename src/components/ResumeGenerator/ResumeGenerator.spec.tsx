import React from 'react';
import { shallow } from 'enzyme';
import ResumeGenerator from './ResumeGenerator';

describe('ResumeGenerator Component', () => {
  const mockFunc = jest.fn();
  const mockedEvent = { preventDefault: () => {} };
  const wrapper = shallow(<ResumeGenerator onSubmit={mockFunc} />);
  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('changes input fields value on change', () => {
    const event = { currentTarget: { value: 'test' } };
    const input = wrapper.find('input');

    input.simulate('change', event);
    expect(wrapper.find('input').prop('value')).toBe('test');
  });

  it('handles button clicks ', () => {
    wrapper.find('button').simulate('click', mockedEvent);
    expect(mockFunc).toHaveBeenCalled();
  });
});
