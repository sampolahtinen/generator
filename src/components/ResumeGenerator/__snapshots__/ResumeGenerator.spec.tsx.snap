// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResumeGenerator Component renders 1`] = `
<div
  className="resumeGenerator"
>
  <h3>
    Github username
  </h3>
  <form
    aria-label="Input github username"
  >
    <label
      className="hidden"
      htmlFor="username"
    >
      Github username input
    </label>
    <input
      autoComplete="off"
      autoFocus={true}
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="John Doe"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
      type="submit"
    >
      Generate
    </button>
  </form>
  <span
    className="highlighted-text"
  />
</div>
`;
